src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPoolManager.cs

```diff
@@ -231,7 +231,7 @@ public Task<HttpResponseMessage> SendAsyncCore(HttpRequestMessage request, Uri p
                 // So, we need to add them manually for now.
                 bool isNonNullIPv6address = key.Host != null && request.RequestUri.HostNameType == UriHostNameType.IPv6;
 
-                pool = new HttpConnectionPool(this, key.Kind, isNonNullIPv6address ? "[" + key.Host + "]" : key.Host, key.Port, key.SslHostName, key.ProxyUri, _maxConnectionsPerServer);
+                pool = new HttpConnectionPool(this, key.Kind, isNonNullIPv6address ? "[" + key.Host + "]" : key.Host, key.Port, key.SslHostName, key.ProxyUri, _maxConnectionsPerServer, request.Headers);
```

Comment:
This is not the right place to add the headers.   We do create a new pool if we don't have one yet.  So, now it will remember the first request to be sent and will remember the headers of the first request.

But for subsequent requests which might not have the 'User-Agent' request header (for example), it won't be doing the right thing.

So, this logic is not correct.

Reply:
And thinking about this more, the best that can be done is to have the first request for any new connection (which thus requires a CONNECT tunnel) to have that specific 'User-Agent' header sent.  In fact, it is not possible to send per-request 'User-Agent' headers because multiple requests using the same connection only send one CONNECT verb.

Reply:
Should we always grab and store all headers even if we only use UserAgent?

Reply:
>Should we always grab and store all headers even if we only use UserAgent?

No.  And as I mentioned in the feedback.  This PR is functionally incorrect.  It's currently always sending the 'User-Agent' header from the very first request for every CONNECT verb (which means every new connection to the proxy).   It shouldn't be doing that.